pkg/console/network.go

```diff
@@ -16,14 +16,26 @@ import (
 	"github.com/harvester/harvester-installer/pkg/config"
 )
 
-func applyNetworks(network config.Network, hostname string) ([]byte, error) {
+func applyNetworks(network config.Network, hostname string, dhclientSetHostname bool) ([]byte, error) {
 	if err := config.RestoreOriginalNetworkConfig(); err != nil {
 		return nil, err
 	}
 	if err := config.SaveOriginalNetworkConfig(); err != nil {
 		return nil, err
 	}
 
+	yesOrNo := "no"
+	if dhclientSetHostname {
+		yesOrNo = "yes"
+	}
+	output, err := exec.Command("sed", "-i",
+		fmt.Sprintf(`s/^DHCLIENT_SET_HOSTNAME=.*/DHCLIENT_SET_HOSTNAME="%s"/`, yesOrNo),
+		"/etc/sysconfig/network/dhcp").CombinedOutput()
```

Comment:
The file control global settings. Should we set the value on the management interface only? Although we don't request IPs on VLAN bridges (those generated from harvester-network-controller), setting this per interface can lower the chance of interfering with other interfaces.

From the [man page](https://manpages.opensuse.org/Tumbleweed/wicked/ifcfg-dhcp.5.en.html) it seems we can override the setting in the `ifcfg-mgmt-br` file. And if we make the change, we can consider moving the configuration field `dhclientSetHostname` into `install.management_interface`.

Reply:
Yeah, I agree, setting this on the management interface only rather than globally makes good sense.